### ISSUE TITLE
Tests with special requirements should be tagged to enable turning them on or off

### ISSUE THREAD

**Issue by bgyori** [sorgerlab/indra#354]
We should define a set of tags for tests, some basic categories being:
- `slow` for tests that are slow, like reading and processing a full paper
- `webservice` for tests that use outside web services
- `nonpublic` for tests that rely on non-public environment variables to run
with possibly other ones that seem useful.

There are several ways to implement this, and the one that to me seems like the best fit is using `nose.plugins.attrib` (http://nose.readthedocs.io/en/latest/plugins/attrib.html), which lets you add a decorator to a test like
```
from nose.plugins.attrib import attr
@attr(speed='slow')
def test_something_slow():
    ...
```
and then run tests skipping a specific tag as
```
nosetests -a '!slow'
```
Given a set of tags, we should review all tests in INDRA and apply tags as needed.

A useful further step will be setting up the Travis configuration to take arguments to `nosetests -a` as environment variables. This would let us define that, for instance, PRs to sorgerlab should not run `nonpublic` tests since hidden environment variables are not enabled in PRs, and these tests always fail.

**Comment by pagreene** [comment]
Would 'nonpublic' environment variables include database urls?

**Comment by bgyori** [comment]
Yes, if the database is not publicly accessible. The point would be to tag everything with 'nonpublic' that would fail in a PR due to missing environmental variables.

**Comment by cthoyt** [comment]
When I write tests that require environment variables, I have nose skip them if they are unavailable
```python
import unittest
import os

class TestStuff(unittest.TestCase):

    @unitttest.skipUnless('CONNECTION' in os.environ, reason='missing connection')
    def test_thing(self):
        ...
```

**Comment by pagreene** [comment]
Issue resolved with #362.
